fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -196,6 +207,9 @@ public class Coordinator {
     private TUniqueId queryId;
     private final ConnectContext connectContext;
     private final boolean needReport;
+    private final boolean preferComputeNode;
+    //this query use compute node number
+    private final int useComputeNodeNumber;
```

Comment:
Do you mind explain why we let users to decide how many compute node to use?

Reply:
We are not sure how many Compute nodes there are in the cluster, and Compute nodes can be dynamically expanded. At a certain moment, a large number of Compute nodes need to be expanded to solve a specific query, but the original query does not actually use too many resources, so This simple query can be executed by specifying a certain number of Compute nodes, which can increase the flexibility of using Compute nodes while saving resources.

Reply:
1. If we are sure each compute node in the cluster, how could we decide which compute node to assigned the task?
2. It depends on the strategy of scale out & in of compute nodes, if it's decided by the resource usage of the whole cluster or the number of compute node is using